src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs

```diff
@@ -274,29 +269,55 @@ internal ResponseData(HttpResponseMessage? response, Guid loggingRequestId, long
             public override string ToString() => $"{{ {nameof(Response)} = {Response}, {nameof(LoggingRequestId)} = {LoggingRequestId}, {nameof(Timestamp)} = {Timestamp}, {nameof(RequestTaskStatus)} = {RequestTaskStatus} }}";
         }
 
-        private static void InjectHeaders(Activity currentActivity, HttpRequestMessage request)
+        private static class Settings
         {
-            const string TraceParentHeaderName = "traceparent";
-            const string TraceStateHeaderName = "tracestate";
-            const string RequestIdHeaderName = "Request-Id";
-            const string CorrelationContextHeaderName = "Correlation-Context";
+            private const string EnableActivityPropagationEnvironmentVariableSettingName = "DOTNET_SYSTEM_NET_HTTP_ENABLEACTIVITYPROPAGATION";
+            private const string EnableActivityPropagationAppCtxSettingName = "System.Net.Http.EnableActivityPropagation";
+
+            public static readonly bool s_activityPropagationEnabled = GetEnableActivityPropagationValue();
+
+            private static bool GetEnableActivityPropagationValue()
+            {
+                // First check for the AppContext switch, giving it priority over the environment variable.
+                if (AppContext.TryGetSwitch(EnableActivityPropagationAppCtxSettingName, out bool enableActivityPropagation))
+                {
+                    return enableActivityPropagation;
+                }
+
+                // AppContext switch wasn't used. Check the environment variable to determine which handler should be used.
+                string? envVar = Environment.GetEnvironmentVariable(EnableActivityPropagationEnvironmentVariableSettingName);
+                if (envVar != null && (envVar.Equals("false", StringComparison.OrdinalIgnoreCase) || envVar.Equals("0")))
+                {
+                    // Suppress Activity propagation.
+                    return false;
+                }
 
+                // Defaults to enabling Activity propagation.
+                return true;
+            }
```

Comment:
Heads up, that I want to move this logic to a shared `GlobalHttpSettings` class, to centralize the process-wide configuration we read from AppContext / environment variables. (See discussion starting with https://github.com/dotnet/runtime/pull/54755#discussion_r660067631.)

Reply:
No worries @antonfirsov - this is just a test PR reverting #54437 to see if timeout failures will reproduce in CI

Reply:
Ah ok, I thought it's a follow-up change.